File: frontend/src/data/officePosts.js
// Canonical office-bearer posts. Stay in sync with
// /app/backend/data/office_posts.py — the backend enforces uniqueness for
// Chairman / Secretary / Treasurer (only one person can hold each). Assistant
// can be held by any number of people.
export const OFFICE_POSTS = [
  "Chairman",
  "Secretary",
  "Treasurer",
  "Assistant",
];

export const UNIQUE_POSTS = new Set(["Chairman", "Secretary", "Treasurer"]);

File: backend/data/office_posts.py
"""Canonical office-bearer posts for Heroic HIFI Foundation.

Only the Master Admin (``is_super_admin``) can attach or remove a post.
Posts in ``UNIQUE_POSTS`` can be held by **at most one** person at a time —
the backend blocks double-assignment with HTTP 400. ``Assistant`` is the
only post that can be held by many people simultaneously.
"""

OFFICE_POSTS: list[str] = [
    "Chairman",
    "Secretary",
    "Treasurer",
    "Assistant",
]

# Posts that may only be held by a single user at any time.
UNIQUE_POSTS: set[str] = {"Chairman", "Secretary", "Treasurer"}

OFFICE_POSTS_SET = set(OFFICE_POSTS)
